components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -107,31 +121,50 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
         // goes to #users if (1) explicit #users, (2) no fragment, or (3) invalid fragment
         this.tabValue = (fragment === 'details') ? 'details' : 'users';
       });
+
+    this.store.select(iamMinorVersion)
+      .pipe(
+        takeUntil(this.isDestroyed),
+        filter(identity)
+      )
+      .subscribe((version) => {
+        this.isMinorV1 = version === 'v1';
+      });
+
+    this.store.select(assignableProjects)
+      .subscribe((assignable: ProjectsFilterOption[]) => {
+        assignable.forEach(({ value: id, label: name, type: stringType }) => {
+          const type = <IAMType>stringType;
```

Comment:
That results in:

```
Type 'string' is not assignable to type 'IAMType'.
```

I'm pretty sure we actually need to cast it to a string here.

Reply:
When the types are compatible @bcmdarroch's suggestion is much preferred. But it does not work here, as @tylercloke noted. You would have to do it like this, which does not add any value over the original (sort of worse off, since it is redundant)
```
const type: IAMType = <IAMType>stringType;
```
So I would suggest go back to the original. 🤷‍♂

Reply:
Where does `stringType` come from? Can we type it at its source?

Also, let's introduce a type guard for `IAMType` -- this way, we'll get an error that can be dealt with instead of somewhat undefined behaviour at runtime. Right now, this is saying _"whatever `stringType` is, dear compiler, assume it's `IAMType`"_. This throws out type safety 😉